src/main/java/lotto/LottoMatchResult.java

```diff
@@ -0,0 +1,35 @@
+package lotto;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class LottoMatchResult {
+
+  private final Map<LottoRank, Integer> matchResult = new HashMap<>();
+
+  public void match(LottoRank rank) {
+    matchResult.put(rank, matchResult.getOrDefault(rank, 0) + 1);
+  }
+
+  public int countMatches(LottoRank rank) {
+    return matchResult.getOrDefault(rank, 0);
+  }
+
+  public long totalPrize() {
+    long totalPrize = 0;
+    for (LottoRank rank : LottoRank.values()) {
+      int count = countMatches(rank);
+      totalPrize += (long) count * rank.prize();
```

Comment:
```suggestion
      totalPrize += rank.prize(count);
```
enum도 일반 자바 클래스와 같은 특성을 가지기 때문에 위와 같이 값을 꺼내기보다 메시지를 보내는 방식으로도 접근 가능함